### PR TITLE
Fix compliance message format bug

### PR DESCRIPTION
**Pull request by JustinKuli**
When specific numbers of users are found with the role, the compliance message was getting shortened incorrectly.

Refs:
 - https://issues.redhat.com/browse/ACM-8411

### CODE REVIEW

**Comment by JustinKuli** [comment]
/hold

I need to write a test

**Review by dhaiducek — COMMENTED** [review]
This works, but I'm wondering about the reliability (mostly because I'm unfamiliar with IamPolicy). Maybe use this function here instead?
https://github.com/stolostron/iam-policy-controller/blob/5c89098348afd5d677350c39a64c4e90b04621f1/controllers/iampolicy_controller.go#L487

**Comment by dhaiducek** [comment]
> This works, but I'm wondering about the reliability (mostly because I'm unfamiliar with IamPolicy). Maybe use this function here instead?
> 
> https://github.com/stolostron/iam-policy-controller/blob/5c89098348afd5d677350c39a64c4e90b04621f1/controllers/iampolicy_controller.go#L487

Actually, for that you'd need to specify a role wildcard and it might be too specific, so maybe simpler is better for this...

File: test/e2e/basic_test.go [hunk 1]
```diff
@@ -10,23 +10,39 @@ import (
...
 				return comp
 			}, defaultTimeoutSeconds, 1).Should(Equal("Compliant"))
+
+			By("Checking parent policy events")
+			compParentEvents := GetMatchingEvents(testNamespace, policyParentName,
+				"policy: "+testNamespace+"/"+policyName, "^Compliant;")
+			Expect(compParentEvents).NotTo(BeEmpty())
+			nonCompParentEvents := GetMatchingEvents(testNamespace, policyParentName,
+				"policy: "+testNamespace+"/"+policyName, "^NonCompliant;")
+			Expect(nonCompParentEvents).To(BeEmpty())
```
Comment (dhaiducek) [inline test/e2e/basic_test.go:29]: Should this be wrapped in `Eventually()`/`Consistently()`?

File: test/e2e/basic_test.go [hunk 2]
```diff
@@ -10,23 +10,39 @@ import (
...
 				pol := GetWithTimeout(gvrIamPolicy, policyName, testNamespace, true)
 				comp, _, _ := unstructured.NestedString(pol.Object, "status", "compliant")
 
 				return comp
 			}, defaultTimeoutSeconds, 1).Should(Equal("NonCompliant"))
+
+			By("Checking parent policy events")
+			nonCompParentEvents := GetMatchingEvents(testNamespace, policyParentName,
+				"policy: "+testNamespace+"/"+policyName, "^NonCompliant;.*"+expectedMsg)
+			Expect(nonCompParentEvents).NotTo(BeEmpty())
```
Comment (dhaiducek) [inline test/e2e/basic_test.go:45]: (Same as above)

**Review by dhaiducek — CHANGES_REQUESTED** [review]

File: test/e2e/basic_test.go (hunk 1 above)
Comment (JustinKuli) [reply test/e2e/basic_test.go:29]: Good idea!

File: Makefile [hunk 3]
```diff
@@ -286,6 +286,7 @@ kind-delete-cluster:
 install-crds:
 	@echo installing crds
 	kubectl apply -f deploy/crds/policy.open-cluster-management.io_iampolicies.yaml
+	kubectl apply -f https://raw.githubusercontent.com/stolostron/governance-policy-propagator/main/deploy/crds/policy.open-cluster-management.io_policies.yaml
```
Comment (dhaiducek) [inline Makefile:289]: I'd say this should have a variable branch specified to replace `main`, but I don't think it really matters for this.

**Review by dhaiducek — APPROVED** [review]
LGTM!

**Pull request merged**